### PR TITLE
Fix #5683: Add support for dynamic type on the url & bottom toolbars

### LINKED ISSUE #5683: Add support for Dynamic Type on the URL bar & toolbar
### Description: 

This is an issue that outlines a small subset of the full dynamic type support issue (#1344)

This should add dynamic font + image sizes on the URL bar, tab bar and toolbar

## Test Plan

- Test that toolbar buttons scale at Large, XL and XXL and get no bigger after
- Test landscape + iPad to ensure bottom toolbar buttons that end up in the top toolbar behave the same
- Test various extra URL bar buttons: playlist, reader mode, wallet, rewards behave the same
- Test that various onboarding popovers that highlight elements in the URL bar look correct
- Test that URL bar text scales until Accessibility Large and no more after
- Test that the collapsed URL bar does the same
- Test that the tab titles in the tab bar scale until XXL and no further

### PR DESCRIPTION
**Pull request by kylehickinson**
## Summary of Changes

This pull request fixes #5683 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:

See issue

## Screenshots:

| XS | Large (Default) | XL | XXL | AL |
| --- | --- | --- | --- | --- |
| Simulator Screen Shot - iPhone 13 Pro - 2022-07-13 at 17 16 28 | Simulator Screen Shot - iPhone 13 Pro - 2022-07-13 at 17 16 35 | Simulator Screen Shot - iPhone 13 Pro - 2022-07-13 at 17 16 38 | Simulator Screen Shot - iPhone 13 Pro - 2022-07-13 at 17 16 39 | Simulator Screen Shot - iPhone 13 Pro - 2022-07-13 at 17 16 41 |

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift [hunk 1]
```diff
@@ -376,7 +410,7 @@ class TabLocationView: UIView {
   }
   
   fileprivate func updateTextWithURL() {
-    (urlTextField as? DisplayTextField)?.hostString = url?.host ?? ""
+    (urlTextField as? DisplayTextField)?.hostString = url?.withoutWWW.host ?? ""
```
Comment (kylehickinson) [inline Client/Frontend/Browser/Toolbars/UrlBar/TabLocationView.swift:413]: This line was actually a bit bugged prior to this change and became obvious during my tests with really large fonts. The host provided was something like `www.amazon.com` but the visible text was `amazon.com`, so the x origin was being altered slightly too much when the two were not matching

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -943,9 +943,9 @@ public class BrowserViewController: UIViewController, BrowserViewControllerDeleg
       make.left.right.equalTo(self.view)
     }
 
-    topToolbar.snp.makeConstraints { make in
-      make.height.equalTo(UIConstants.topToolbarHeight)
-    }
+//    topToolbar.snp.makeConstraints { make in
+//      make.height.equalTo(UIConstants.topToolbarHeight)
+//    }
```
Comment (iccub) [inline Client/Frontend/Browser/BrowserViewController.swift]: This comment still relevant?

File: Client/Frontend/Browser/Toolbars/UrlBar/RewardsButton.swift [hunk 3]
```diff
@@ -23,33 +23,39 @@ class RewardsButton: UIButton {
...
     accessibilityLabel = Strings.rewardsPanel
     accessibilityIdentifier = "urlBar-rewardsButton"
 
     addSubview(lookAtMeBadge)
 
     updateView()
+    
+    lookAtMeBadge.snp.makeConstraints {
+      $0.top.equalToSuperview()
+      $0.leading.equalTo(imageView!.snp.centerX)
```
Comment (iccub) [inline Client/Frontend/Browser/Toolbars/UrlBar/RewardsButton.swift:44]: this force unwrap safe here?

File: Client/Frontend/Browser/BrowserViewController.swift (hunk 2 above)
Comment (kylehickinson) [reply Client/Frontend/Browser/BrowserViewController.swift]: Nope, accidentally left the comment here, will delete

File: Client/Frontend/Browser/Toolbars/UrlBar/RewardsButton.swift (hunk 3 above)
Comment (kylehickinson) [reply Client/Frontend/Browser/Toolbars/UrlBar/RewardsButton.swift:44]: Yep, this value is only nil for system buttons as per UIButton docs

**Review by StephenHeaps — APPROVED** [review]

File: Sources/BraveUI/Extensions/UITraitCollectionExtensions.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  }
+  
+  private func _clampingSizeCategory(
+    minimum: UIContentSizeCategory? = nil,
+    maximum: UIContentSizeCategory? = nil
+  ) -> Self {
+    let minCategory = minimum ?? .extraSmall
+    let maxCategory = maximum ?? .accessibilityExtraExtraExtraLarge
+    let clampedCategory = preferredContentSizeCategory < minCategory ? minCategory : preferredContentSizeCategory > maxCategory ? maxCategory : preferredContentSizeCategory
+    return .init(preferredContentSizeCategory: clampedCategory)
```
Comment (Brandon-T) [inline Sources/BraveUI/Extensions/UITraitCollectionExtensions.swift:35]: `max(minCategory, min(preferredContentSizeCategory, maxCategory))`
will work?

**Review by Brandon-T — CHANGES_REQUESTED** [review]

File: Sources/BraveUI/Extensions/UITraitCollectionExtensions.swift (hunk 4 above)
Comment (kylehickinson) [reply Sources/BraveUI/Extensions/UITraitCollectionExtensions.swift:35]: No unfortunately `UIContentSizeCategory` does not conform to `Comparable`, despite having `<` and `>` static functions for comparison 😅

**Review by iccub — APPROVED** [review]
All functional questions I had are resolved

**Review by Brandon-T — APPROVED** [review]

**Pull request merged**